tests/epyccel/test_builtins.py

```diff
@@ -1,41 +1,152 @@
-# pylint: disable=missing-function-docstring, missing-module-docstring/
-
-from pyccel.epyccel import epyccel
-from pyccel.decorators import types
-
-def test_abs_i(language):
-    @types('int')
-    def f1(x):
-        return abs(x)
-
-    f2 = epyccel(f1, language=language)
-
-    assert f1(0) == f2(0)
-    assert f1(-5) == f2(-5)
-    assert f1(11) == f2(11)
-
-def test_abs_r(language):
-    @types('real')
-    def f1(x):
-        return abs(x)
-
-    f2 = epyccel(f1, language=language)
-
-    assert f1(0.00000) == f2(0.00000)
-    assert f1(-3.1415) == f2(-3.1415)
-    assert f1(2.71828) == f2(2.71828)
-
-
-
-def test_abs_c(language):
-    @types('complex')
-    def f1(x):
-        return abs(x)
-
-    f2 = epyccel(f1, language=language)
-
-    assert f1(3j + 4) == f2(3j + 4)
-    assert f1(3j - 4) == f2(3j - 4)
-    assert f1(5j + 0) == f2(5j + 0)
-    assert f1(0j + 5) == f2(0j + 5)
-    assert f1(0j + 0) == f2(0j + 0)
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+import pytest
+from numpy.random import randint
+
+from pyccel.epyccel import epyccel
+from pyccel.decorators import types, template
+
+def test_abs_i(language):
+    @types('int')
+    def f1(x):
+        return abs(x)
+
+    f2 = epyccel(f1, language=language)
+
+    assert f1(0) == f2(0)
+    assert f1(-5) == f2(-5)
+    assert f1(11) == f2(11)
+
+def test_abs_r(language):
+    @types('real')
+    def f1(x):
+        return abs(x)
+
+    f2 = epyccel(f1, language=language)
+
+    assert f1(0.00000) == f2(0.00000)
+    assert f1(-3.1415) == f2(-3.1415)
+    assert f1(2.71828) == f2(2.71828)
+
+
+
+def test_abs_c(language):
+    @types('complex')
+    def f1(x):
+        return abs(x)
+
+    f2 = epyccel(f1, language=language)
+
+    assert f1(3j + 4) == f2(3j + 4)
+    assert f1(3j - 4) == f2(3j - 4)
+    assert f1(5j + 0) == f2(5j + 0)
+    assert f1(0j + 5) == f2(0j + 5)
+    assert f1(0j + 0) == f2(0j + 0)
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="min not implemented in C"),
+            pytest.mark.c]
```

Comment:
the min function of two values is available in c.

Reply:
Can you provide a reference?`It exists in c++ and certain compilers but I don't think it's in the standard. See https://stackoverflow.com/questions/3437404/min-and-max-in-c

Reply:
sorry I mean float just in include `math.h`, it exists for integers only on a few compilers http://tigcc.ticalc.org/doc/stdlib.html

Reply:
Fixed in 7c2a879.
Tests fixed in 348be83

We shouldn't implement things that are compiler specific. I have just implemented it for 2 float arguments

Reply:
ok great.